src/Abp.ZeroCore.EntityFrameworkCore/EntityHistory/EntityHistoryHelper.cs

```diff
@@ -382,6 +382,22 @@ private void UpdateChangeSet(EntityChangeSet changeSet)
                 var entityEntry = entityChange.EntityEntry.As<EntityEntry>();
                 entityChange.EntityId = GetEntityId(entityEntry);
 
+                /* Update audit field */
+
+                foreach (var propertyChange in entityChange.PropertyChanges.
```

Comment:
are the audit fields being set after `SaveChanges`?
if that is the case, `entityChange.PropertyChanges` should not contain the audit fields as `entityChange.PropertyChanges` are generated before `SaveChanges`

Reply:
https://github.com/aspnetboilerplate/aspnetboilerplate/blob/41548c2927ca1b8cebd75612c4375602373be19b/src/Abp.ZeroCore.EntityFrameworkCore/EntityHistory/EntityHistoryHelper.cs#L213

ShouldSavePropertyHistory is true when creating an entity.
see: https://github.com/aspnetzero/aspnet-zero-core/issues/2490

Reply:
Audit fields are having `ShouldSavePropertyHistory = true` because the `Blog` class is having `AuditedAttribute`.

Also, not sure if the bug is accurate, base on 
https://github.com/aspnetboilerplate/aspnetboilerplate/blob/d95f52a0e950ea51f5ddffb274ebedf52d6daef7/src/Abp.EntityFrameworkCore/EntityFrameworkCore/AbpDbContext.cs#L203-L216

`ApplyAbpConcepts()` which sets the CreatorUserId is called before `base.SaveChangesAsync()` and this contradict with the assumption that `CreatorUserId` not captured by `EntityHistory` due to assignment of User ID after `SaveChanges`

Reply:
https://github.com/aspnetboilerplate/aspnetboilerplate/blob/cfa75bddf0ac0a41d74300162ed50becc2a926c2/src/Abp.Zero.EntityFramework/Zero/EntityFramework/AbpZeroCommonDbContext.cs#L197
First execute` EntityHistoryHelper?.CreateEntityChangeSet(this)`,
The code then goes to the `ApplyAbpConcepts` method in the base class `SaveChanges`.

Reply:
Right. I missed that 😅